git_theta/updates/base.py

```diff
@@ -0,0 +1,68 @@
+"""Base class for parameter update plugins."""
+
+import os
+import sys
+
+if sys.version_info < (3, 10):
+    from importlib_metadata import entry_points
+else:
+    from importlib.metadata import entry_points
+
+import logging
+from typing import Optional
+
+from git_theta import git_utils, utils, params, metadata
+
+
+class Update:
```

Comment:
I have some ideas on how to refactor the Update class to be more encapsulated (and it should help making things async eventually) but those are big changes. We should punt until this is landed.